res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,980 @@
+///////////////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                                          //
+///////////////////////////////////////////////////////////////////////////////////
+/* global engine                                                                 */
+/* global HIDDebug                                                               */
+/* global HIDPacket                                                              */
+/* global HIDController                                                          */
+/* jshint -W016                                                                  */
+///////////////////////////////////////////////////////////////////////////////////
+/*                                                                               */
+/* Traktor Kontrol S2 MK3 HID controller script v1.00                            */
+/* Last modification: October 2019                                               */
+/* Author: Michael Schmidt                                                       */
+/* https://www.mixxx.org/wiki/doku.php/native_instruments_traktor_kontrol_s2_mk3 */
+/*                                                                               */
+///////////////////////////////////////////////////////////////////////////////////
+
+TraktorS2MK3 = function () {
+    this.controller = new HIDController();
+    this.shiftPressed = { "[Channel1]": false, "[Channel2]": false };
+    this.browseState = { "[Channel1]": 0, "[Channel2]": 0 };
+    this.loopSizeState = { "[Channel1]": 0, "[Channel2]": 0 };
+    this.beatjumpState = { "[Channel1]": 0, "[Channel2]": 0 };
+    this.fxButtonState = { 1: false, 2: false, 3: false, 4: false };
+    this.padModeState = { "[Channel1]": 0, "[Channel2]": 0 }; // 0 = Hotcues Mode, 1 = Samples Mode
+
+    // Microphone button
+    this.microphonePressedTimer = 0; // Timer to distinguish between short and long press
+
+    // Sync buttons
+    this.syncPressedTimer = { "[Channel1]": 0, "[Channel2]": 0 }; // Timer to distinguish between short and long press
+
+    // Jog wheels
+    this.lastTickVal = [0, 0];
+    this.lastTickTime = [0.0, 0.0];
+
+    // VuMeter
+    this.vuLeftConnection = {};
+    this.vuRightConnection = {};
+    this.clipLeftConnection = {};
+    this.clipRightConnection = {};
+    this.vuMeterThresholds = { "vu-18": (1 / 6), "vu-12": (1 / 6 * 2), "vu-6": (1 / 6 * 3), "vu0": (1 / 6 * 4), "vu6": (1 / 6 * 5) };
+
+    // Sampler callbacks
+    this.samplerCallbacks = [];
+    this.samplerHotcuesRelation = {
+        "[Channel1]": {
+            1: 1, 2: 2, 3: 3, 4: 4, 5: 9, 6: 10, 7: 11, 8: 12
+        }, "[Channel2]": {
+            1: 5, 2: 6, 3: 7, 4: 8, 5: 13, 6: 14, 7: 15, 8: 16
+        }
+    };
+};
+
+var TraktorS2MK3 = new TraktorS2MK3();
```

Comment:
I'm not sure if it's a good idea to overwrite a class with an instance of itself. This might work, but could break in the future. Maybe just chose another name?

Reply:
You are right, i will change this back to this: `var TraktorS2MK3 = new function () { ... }`. I separated class definition and instantiation because of the code checking tools giving warnings here. Since there is only one instance of the class this might be not as confusing as using a different name.